src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1120,8 +1122,15 @@ private ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedA
                                         newArguments[i] = ((UnboundLambda)oldArgument).Bind(parameterType);
                                     }
                                     break;
+                                case BoundKind.OutDeconstructVarPendingInference:
+                                    newArguments[i] = ((OutDeconstructVarPendingInference)oldArgument).SetInferredType(parameters[i].Type, success: true);
+                                    break;
                             }
                         }
+                        else if (oldArgument.Kind == BoundKind.OutDeconstructVarPendingInference)
+                        {
+                            newArguments[i] = ((OutDeconstructVarPendingInference)oldArgument).FailInference(this);
```

Comment:
We probably need to have similar code in BuildArgumentsForDynamicInvocation, or at least an assert that we cannot run into OutDeconstructVarPendingInference there.